pages/chat/[sender]/[receiver].js

```diff
@@ -0,0 +1,186 @@
+import SendMessage from "@/components/Chat/SendMessage";
+
+export default function Chat() {
+  const router = useRouter();
+  const { sender, receiver } = router.query;
+  const messageLastRef = useRef(null)
+  const [chats, setChats] = useState([]);
+  const [messageToSend, setMessageToSend] = useState("");
+  const { data: session } = useSession();
+  const userName = session?.user.name;
+
+  const output = sender?.toLocaleLowerCase();
+  const input = receiver?.toLocaleLowerCase();
+
+  useEffect(() => {
+    Pusher.logToConsole = true;
+
+    const pusher = new Pusher( process.env.NEXT_PUBLIC_PUSHER_KEY , {
+      cluster: 'eu'
+    });
+
+    const channel = pusher.subscribe('chat');
+
+    channel.bind("chatMessages", function (data) {
+
+      console.log('this is sender', data.sender);
+      console.log('this is receiver', data.receiver);
+
+      if (data.sender === output && data.receiver === input || data.sender === input && data.receiver === output) {
+        setChats((prevState) => [
+          ...prevState,
+          { sender: data.sender, message: data.message, receiver: data.receiver },
+        ]);
+      }
+    });
+
+    return () => {
+      pusher.unsubscribe("chat");
+    };
+
+  }, [receiver]);
+
+  useEffect(() => {
+    messageLastRef.current?.scrollIntoView({ behavior: "smooth" });
+  },[chats])
+
+  if (!userName) return <h2>Loading ...</h2>
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+
+    try {
+      await fetch("/api/chat", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ message: messageToSend, sender: output, receiver: input }),
+      });
+
+      await mutate;
+
+    } catch (error) {
+      console.error("Fehler beim Senden der Nachricht:", error);
+    }
+
+    setMessageToSend('')
+  }
+
+  return (
+    <StyledChat>
+      <ChatContent>
+        <ChatMessages>
+          {chats.map((chat, id) => (
+            <Message key={ id } variant={chat.sender === output ? 'current' : 'other' }>
+              <Sender>{ chat.sender }</Sender>
+              <Bubble>{ chat.message }</Bubble>
+            </Message>
+          ))}
+
+          <div ref={messageLastRef}></div>
+        </ChatMessages>
+        <SendMessage
+          message={messageToSend}
+          handleMessageChange={(event) => setMessageToSend(event.target.value)}
+          handleSubmit={(event) => {
+            handleSubmit(event)
+          }}/>
+      </ChatContent>
+    </StyledChat>
+  );
+}
+
+const StyledChat = styled.div`
+  display: flex;
+  align-items: stretch;
+  width: 100vw;
+  height: 100vh;
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+`;
+
+const ChatContent = styled.div`
+  flex: 0 0 100%;
+  width: 100%;
+  background: #5b6465;
+  padding: 55px 0 60px;
+  display: flex;
+  align-items: stretch;
+  flex-direction: column;
+`;
+
+const ChatMessages = styled.div`
```

Comment:
I would expect this to be a list, because inside of it, you map over an array which should create list items.

Reply:
Why? The messages has no information background, nothing there it is necessary to shown as list with marker or something else. Chat messages even not relevant for seo, thats only a comunication between persons.
html elements should always be used for their purpose. In this case, it would be a misappropriation and subject to additional adjustments

Reply:
But what about a person using a screen reader and want to go to the third last message? This would only be possible if the screen reader can figure out how many items there are which would be possible with a list.

Besides, you can (and should) make the bullet points invisible so that the HTML list does not affect your goal.